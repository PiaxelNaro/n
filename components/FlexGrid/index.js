// @flow
import styled, { css } from 'styled-components';

const FlexGrid = styled.div`
    display: ${props => (props.inline ? 'inline-flex' : 'flex')};
    align-items: center;
    ${props => (props.gutter ? css`
        & > :nth-child(n + 2) {
            margin-left: ${props.gutter};
        }
    ` : '')}
`;
FlexGrid.displayName = 'FlexGrid';

FlexGrid.Item = styled.div`
    min-width: ${props => (props.flex > 0 ? 0 : 'auto')};
    flex-shrink: ${props => (props.flex > 0 ? 1 : 0)};
    flex: ${props => props.flex};
    flex-basis: auto;
`;
FlexGrid.Item.defaultProps = { flex: 1 };
FlexGrid.Item.displayName = 'FlexGrid.Item';

const ColumnGrid = styled.div`
    & > :nth-child(n + 2) {
        margin-top: ${props => props.gutter};
    }
`;
export default FlexGrid;
export { ColumnGrid };
