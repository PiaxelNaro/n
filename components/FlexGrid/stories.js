// @flow
import React from 'react';
import { storiesOf } from '@kadira/storybook';
import SearchIcon from 'react-icons/lib/md/search';
import FlexGrid from '.';

// eslint-disable-next-line no-bitwise, prefer-template
const randomColor = () => '#' + (((1 << 24) * Math.random()) | 0).toString(16);
const boxStyle = () => ({
    backgroundColor: randomColor(),
    height: '60px',
});

storiesOf('FlexGrid', module)
    .add('Layout examples', () => (
        <div>
            <FlexGrid>
                <FlexGrid.Item style={boxStyle()} />
                <FlexGrid.Item style={boxStyle()} />
                <FlexGrid.Item style={boxStyle()} />
            </FlexGrid>
            <FlexGrid style={{ marginTop: '20px' }}>
                <FlexGrid.Item style={boxStyle()} />
                <FlexGrid.Item flex={2} style={boxStyle()} />
                <FlexGrid.Item style={boxStyle()} />
            </FlexGrid>
            <FlexGrid style={{ marginTop: '20px' }}>
                <FlexGrid.Item flex={3} style={boxStyle()} />
                <FlexGrid.Item style={boxStyle()} />
            </FlexGrid>
        </div>
    ))
    .add('Grid with Gutter', () => (
        <div>
            <FlexGrid gutter="8px">
                <FlexGrid.Item style={boxStyle()} />
                <FlexGrid.Item style={boxStyle()} />
                <FlexGrid.Item style={boxStyle()} />
                <FlexGrid.Item style={boxStyle()} />
                <FlexGrid.Item style={boxStyle()} />
            </FlexGrid>
            <FlexGrid gutter="20px" style={{ marginTop: '20px' }}>
                <FlexGrid.Item style={boxStyle()} />
                <FlexGrid.Item flex={2} style={boxStyle()} />
                <FlexGrid.Item style={boxStyle()} />
            </FlexGrid>
        </div>
    ))
    .add('Inline alignment', () => (
        <FlexGrid inline>
            <FlexGrid.Item>Loading</FlexGrid.Item>
            <FlexGrid.Item flex={0}>
                <SearchIcon size={40} />
            </FlexGrid.Item>
        </FlexGrid>
    ));
