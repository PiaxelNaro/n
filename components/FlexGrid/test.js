// @flow
import React from 'react';
import { shallow } from 'enzyme';
import FlexGrid, { ColumnGrid } from '.';

describe('<FlexGrid>', () => {
    it('renders inline or as block', () => {
        expect(shallow(<FlexGrid />)).toMatchSnapshot();
        expect(shallow(<FlexGrid inline />)).toMatchSnapshot();
    });
});

describe('<FlexGrid.Item>', () => {
    it('adapt css to flex values', () => {
        expect(shallow(<FlexGrid.Item flex={0} />)).toMatchSnapshot();
        expect(shallow(<FlexGrid.Item flex={1} />)).toMatchSnapshot();
        expect(shallow(<FlexGrid.Item flex={2} />)).toMatchSnapshot();
    });
});

describe('<ColumnGrid />', () => {
    it('setup a vertical spacing grid', () => {
        expect(shallow(<ColumnGrid gutter="8px" />)).toMatchSnapshot();
    });
});
