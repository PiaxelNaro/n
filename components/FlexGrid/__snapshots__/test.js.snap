// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ColumnGrid /> setup a vertical spacing grid 1`] = `
<div
  className="FlexGrid__ColumnGrid-auw7uc-2 kUCwEq"
/>
`;

exports[`<FlexGrid.Item> adapt css to flex values 1`] = `
<div
  className="FlexGrid__Item-auw7uc-1 hroMdl"
/>
`;

exports[`<FlexGrid.Item> adapt css to flex values 2`] = `
<div
  className="FlexGrid__Item-auw7uc-1 fgWlbG"
/>
`;

exports[`<FlexGrid.Item> adapt css to flex values 3`] = `
<div
  className="FlexGrid__Item-auw7uc-1 eMNDdg"
/>
`;

exports[`<FlexGrid> renders inline or as block 1`] = `
<div
  className="FlexGrid__FlexGrid-auw7uc-0 bHHZdg"
/>
`;

exports[`<FlexGrid> renders inline or as block 2`] = `
<div
  className="FlexGrid__FlexGrid-auw7uc-0 kjnORm"
/>
`;
